pin/pin.go

```diff
@@ -285,32 +288,44 @@ func (p *pinner) Flush() error {
 
 	root := &mdag.Node{}
 	{
-		n, err := storeSet(ctx, p.dserv, p.directPin.GetKeys(), recordInternal)
+		n, err := storeSet(ctx, p.statedag, p.directPin.GetKeys(), recordInternal)
```

Comment:
Ahhhh (ooof CRing back and forth on different bases). Wait, i thought we wanted this in 0.3.8 before... ok nvm.